# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/MacroAssembler.cpp
# Commit: 662cb63c42d7
# Full Hash: 662cb63c42d7874a79256a6edf1a03d83e43a014
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2023-08-05 09:19:01
# Regressor Bug: 1692065
# File Overlap Count: 2
# Description:
#   Bug 1692065 - wasm: Create a unique wasm::AnyRef type. r=yury,jonco
#   
#   Right now wasm references area represented as a JSObject*. This is assumed in
#   many places in SpiderMonkey, most prominently in our GC integration
#   around tracing, the store buffer, and write barriers. This will cause issues
# ==============================================================================

diff -r 8b506ab41451 -r 662cb63c42d7 js/src/jit/MacroAssembler.cpp
--- a/js/src/jit/MacroAssembler.cpp	Fri Aug 04 13:54:57 2023 +0000
+++ b/js/src/jit/MacroAssembler.cpp	Fri Aug 04 14:06:49 2023 +0000
@@ -5763,7 +5763,7 @@
     unboxGCThingForGCBarrier(Address(PreBarrierReg, 0), temp1);
   } else {
     MOZ_ASSERT(type == MIRType::Object || type == MIRType::String ||
-               type == MIRType::Shape);
+               type == MIRType::Shape || type == MIRType::RefOrNull);
     loadPtr(Address(PreBarrierReg, 0), temp1);
   }
 
@@ -5781,7 +5781,7 @@
 
   // If the GC thing is in the nursery, we don't need to barrier it.
   if (type == MIRType::Value || type == MIRType::Object ||
-      type == MIRType::String) {
+      type == MIRType::String || type == MIRType::RefOrNull) {
     branchPtr(Assembler::NotEqual, Address(temp2, gc::ChunkStoreBufferOffset),
               ImmWord(0), noBarrier);
   } else {